sources/procfs.go

```diff
@@ -565,18 +510,69 @@ func getJobNum(jobBlock string) (jobID string, err error) {
 	return strings.Trim(jobID, " "), nil
 }
 
-func parseJobStatsText(jobStats string, promName string, helpText string) (metricList []lustreJobsMetric, err error) {
+func getJobStatsOperationMetrics(jobBlock string, jobID string, promName string, helpText string) (metricList []lustreJobsMetric, err error) {
+	operationSlice := []multistatParsingStruct{
+		{index: 0, pattern: "open"},
+		{index: 0, pattern: "close"},
+		{index: 0, pattern: "mknod"},
+		{index: 0, pattern: "link"},
+		{index: 0, pattern: "unlink"},
+		{index: 0, pattern: "mkdir"},
+		{index: 0, pattern: "rmdir"},
+		{index: 0, pattern: "rename"},
+		{index: 0, pattern: "getattr"},
+		{index: 0, pattern: "setattr"},
+		{index: 0, pattern: "getxattr"},
+		{index: 0, pattern: "setxattr"},
+		{index: 0, pattern: "statfs"},
+		{index: 0, pattern: "sync"},
+		{index: 0, pattern: "samedir_rename"},
+		{index: 0, pattern: "crossdir_rename"},
+		{index: 0, pattern: "punch"},
+		{index: 0, pattern: "destroy"},
+		{index: 0, pattern: "create"},
+		{index: 0, pattern: "get_info"},
+		{index: 0, pattern: "set_info"},
+		{index: 0, pattern: "quotactl"},
+	}
+	for _, operation := range operationSlice {
+		opStat := regexCaptureString(operation.pattern+": .*", jobBlock)
+		opNumbers := regexCaptureStrings("[0-9]*.[0-9]+|[0-9]+", opStat)
```

Comment:
Since you aren't using raw string literals here, don't forget the double backslash before the dot since this will match _any_ character between two numbers (e.g. `1234a5678` will return `1234a5678` instead of `1234, 5678`). The pattern will look something like this: `[0-9]*\\.[0-9]+|[0-9]+`.

As an aside, I will cleanup and move all of these number parsers in a future PR, so it isn't too big of a deal.